src/RCall.jl

```diff
@@ -29,13 +30,24 @@ export RObject,
    robject, rcopy, rparse, rprint, reval, rcall, rlang,
    rimport, @rimport, @rlibrary, @rput, @rget, @var_str, @R_str
 
-const depfile = joinpath(dirname(@__FILE__),"..","deps","deps.jl")
-if isfile(depfile)
-    include(depfile)
+@static if @has_preference("Rhome") || @has_preference("libR")
+    @static if !(@has_preference("Rhome") && @has_preference("libR"))
+        error("RCall: Either both Rhome and libR must be set or neither of them")
+    end
+    const Rhome = @load_preference("Rhome")
+    const libR = @load_preference("libR")
+    const conda_provided_r = false
 else
-    error("RCall not properly installed. Please run Pkg.build(\"RCall\")")
+    const depfile = joinpath(dirname(@__FILE__),"..","deps","deps.jl")
+    if isfile(depfile)
+        include(depfile)
+    else
+        error("RCall not properly installed. Please run Pkg.build(\"RCall\")")
+    end
 end
 
+__precompile__(Rhome != "")
```

Comment:
it's very rare to see explicit `__precompile__` directives nowadays -- what happens if you don't have this?

Reply:
This is to abort precompilation when there is not R available so that it is possible to install RCall.jl without an error when there is no R available, in case the user wants to set up preferences between installation and RCall import time. The reason for aborting precompilation is actually more about the dependent packages, since when they are precompiled they will import RCall and `__init__` will run and not be able to complete. In case this behaviour is surprising, see: https://github.com/frankier/precompiletest

I've added a message to this effect and re-written the code to make it more clear that this is about aborting precompilation rather than the old "opting-in" to precompilation approach.